src/gravel/controllers/orch/tests/test_ceph.py

```diff
@@ -1,15 +1,85 @@
+# project aquarium's backend
+# Copyright (C) 2021 SUSE, LLC.
+
+import json
+from typing import Any, Dict
 import pytest
-from pyfakefs.fake_filesystem_unittest import TestCase
+from pyfakefs.fake_filesystem_unittest \
+    import TestCase  # pyright: reportMissingTypeStubs=false
+import unittest
+import unittest.mock
 
 from gravel.controllers.orch.ceph import Mgr, Mon
+from .outputs import mon_df_raw, mon_osdmap_raw
 
 
 class TestCeph(TestCase):
 
     def setUp(self):
-        self.setUpPyfakefs()
+        self.setUpPyfakefs()  # type: ignore
 
     def test_ceph_conf(self):
         with pytest.raises(FileNotFoundError, match="ceph.conf"):
             Mgr()
             Mon()
+
+
+def _exists_conf(cls, confpath: str) -> bool:  # type: ignore
+    return True
+
+
+class MockRados(unittest.mock.MagicMock):
+    pass
+
+
+@unittest.mock.patch(
+    "gravel.controllers.orch.ceph.Ceph._exists_conf",
+    _exists_conf)  # type: ignore
+@unittest.mock.patch("rados.Rados", MockRados)
+class TestMon(unittest.TestCase):
```

Comment:
I have a few problems here I've not been able to overcome:
1. relying on pyfakefs does not seem to make away with the conf file not being present (but I'm sure that's because I'm doing something wrong); and, most importantly,
2. I need to extend from pyfakefs's `TestCase`, which apparently does not play nice with `unittest.mock.patch()`, and I end up having issues with `rados.Rados` not being mocked.

Reply:
https://github.com/aquarist-labs/aquarium/pull/206/commits/1c39879788d0be389c6237311deb5fb7efbe41f4 seems to work ok after merging with this PR and removing the mocks?

Reply:
ack

Reply:
rebased to include #206